language/move-analyzer/src/vfs.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) The Diem Core Contributors
```

Comment:
I wonder what your thoughts are here on using the [`vfs` crate](https://docs.rs/vfs/0.5.1/vfs/) for this instead. It's definitely something I've been considering for the packages work, and I'm wondering if that might make it easier to stitch up here. 

Either way, it's not critical for this PR, but may be worth something considering for the future or discussing :)

Reply:
I would love to use an already-written dependency instead! I was hoping to some day soon convince the rust-analyzer developers to publish [their `vfs` crate](https://github.com/rust-analyzer/rust-analyzer/blob/fdd49b97133a77ff652017d24b72ced37d4a822d/crates/vfs/Cargo.toml).

One thing that deterred me from using the [`vfs` crate you mentioned](https://docs.rs/vfs/0.5.1/vfs/) is that its docs describe its `MemoryFS` as "an ephemeral in-memory implementation **(intended for unit tests)**," but I hadn't looked into it beyond reading that.

Either way, I definitely agree that this code should be deleted in favor of an existing crate; I'd love to discuss which one and why, and I'd also be down to do all this as a follow-up.

Reply:
Definitely not a blocker for this PR! But would definitely be down to discuss -- I'll book us some time next week :) 

`OverlayFS` in the vfs crate might also be interesting since that seems like it could provide some benefits maybe -- I'm thinking you could keep a small in-memory set of files that are actively being worked on and then fetch files on-demand from the underlying FS.  

In general and this is a _much_ larger task, I think it would be nice to parametrize all of our Move code by a VFS trait of some flavor so that we can easily work in-memory or from the disk.